tests/test_smiles_parser.py

```diff
@@ -435,3 +437,40 @@ def metal_in_smiles(smiles):
     assert metal_in_smiles(smiles='CC[W]')
     assert metal_in_smiles(smiles='C[Pd]')
     assert metal_in_smiles(smiles='[Fe3+]CNO[W]')
+
+
+def test_lots_of_smiles_rings():
+
+    parser = Parser()
+
+    # Should be able to parse a SMILES with ring closures with multiple
+    # digits
+    parser.parse(smiles='C%99CCCC%99')
+    cyclopentane_atoms = deepcopy(parser.atoms)
+
+    parser.parse(smiles='C1CCCC1')
+    assert all(parser.atoms[i].label == cyclopentane_atoms[i].label
+               for i in range(len(parser.atoms)))
+
+
+def test_parse_ring_idx():
```

Comment:
you may want to use pytest fixtures here, basically you can run the same test with a different input

Reply:
My 30 second glance wasn't enlightening, but hopefully it's now better?!

Reply:
ah sorry, I siad completely the wrong thing here. I meant parameterize (although fixtures are useful too)
something like
`@pytest.mark.parametrize(
    smiles,
    [
        pytest.param(['C%9CC'], id="name the test"),
        pytest.param(['C%'], id="name this test as well)
    ]
)
def test_parse_ring_idx(smiles):
    with pytest.raises(InvalidSmilesString):
        Parser().parse(smiles)`

Reply:
tbh what you did initially was fine, I probably prefer it to what you changed it to 🙃

Reply:
`mark.parametrize` looks completely unreadable to me, but hey ho – each to their own. Will merge if you're happy

Reply:
go for it